### DISCUSSION TITLE
Correct way to fetch mouse position

### DISCUSSION THREAD

**Discussion by liripoop** [bevyengine/bevy#7970] [Q&A] [answered]
Since bevy 0.9.0 bevy::window:Window does not implement the Resource trait anymore so [this](https://bevy-cheatbook.github.io/cookbook/cursor2world.html) cookbook example does not work anymore. What's the cleanest way to access the mouse position with bevy 0.10.0 and convert it to world space?

**Comment by tim-blackbird** [comment, marked as answer]
`Window` is now a component you need to query for.
I've updated the system to 0.10 for you :)
```rust
fn my_cursor_system(
    windows: Query<&Window>,
    camera_q: Query<(&Camera, &GlobalTransform), With<MainCamera>>,
) {
    let window = windows.single();
    let (camera, camera_transform) = camera_q.single();

    if let Some(world_position) = window
        .cursor_position()
        .and_then(|cursor| camera.viewport_to_world_2d(camera_transform, cursor))
    {
        eprintln!("World coords: {}/{}", world_position.x, world_position.y);
    }
}
```

**Reply by liripoop** [reply]
Tysm thats exactly what i was looking for.

**Comment by tcanabrava** [comment]
For those (like me) that got stuck here, on bevy 0.15 this is now a Result, and not an Option.
so the code would be like:

```
  .and_then(|cursor| {
            let err = camera.viewport_to_world_2d(camera_transform, cursor);
            match err {
                Ok(pos) => Some(pos),
                Err(_) => None,
            }
        })
```

Sorry for the necro bump

**Reply by tim-blackbird** [reply]
For brevity you can use `.ok()`:
```rust
.and_then(|cursor| camera.viewport_to_world_2d(camera_transform, cursor).ok())
```
